### ISSUE TITLE
Long lines in lessons

### ISSUE THREAD

**Issue by maxim-belkin** [swcarpentry/python-novice-inflammation#498]
We need to shorten the following long lines in lessons (based on `make lesson-check-all`):

 - [ ] ./CONDUCT.md: Internally-defined links may be missing definitions: "Carpentry Code of Conduct"=>"coc", "reporting guidelines"=>"coc-reporting"
 - [ ] ./README.md: Line(s) are too long: 1, 2, 8
 - [ ] ./_episodes/01-numpy.md: Line(s) are too long: 32, 39, 40, 49, 140, 160, 175, 177, 178, 199, 227, 311, 322, 323, 345, 399, 560, 766, 820
 - [ ] ./_episodes/02-loop.md: Line(s) are too long: 115, 148, 149, 154, 160, 161, 180
 - [ ] ./_episodes/03-lists.md: Line(s) are too long: 114, 115, 118, 119, 120, 123, 155, 156, 175, 250, 269, 270, 287, 351, 451, 468, 525, 553
 - [ ] ./_episodes/04-files.md: Line(s) are too long: 36, 46
 - [ ] ./_episodes/05-cond.md: Line(s) are too long: 96, 314, 368
 - [ ] ./_episodes/06-func.md: Line(s) are too long: 46, 69, 229, 358, 371, 406, 409, 426, 502, 519, 542, 621, 710, 927, 999
 - [ ] ./_episodes/07-errors.md: Line(s) are too long: 58, 80, 86, 110, 193, 240, 284, 386, 480, 537
 - [ ] ./_episodes/08-defensive.md: Line(s) are too long: 52, 95, 97, 99, 102, 114, 151, 182, 459, 469
 - [ ] ./_episodes/10-cmdline.md: Line(s) are too long: 66, 67, 145, 312, 596, 760, 898, 926, 941
 - [ ] ./_extras/discuss.md: Line(s) are too long: 57, 62, 148
 - [ ] ./index.md: Line(s) are too long: 26, 58
 - [ ] ./reference.md: Line(s) are too long: 23, 45, 64, 161, 180, 247
 - [ ] ./setup.md: Line(s) are too long: 8, 24

**Comment by doctornerdis** [comment]
Just for my own learning, why is it important to limit the lines to 100 characters? Is it about readability of the code? Are there other formatting conventions I should keep in mind?

**Comment by doctornerdis** [comment]
Also, please ignore #519. I forgot to delete my old fork before I started work on the new file.

**Comment by maxim-belkin** [comment]
> Just for my own learning, why is it important to limit the lines to 100 characters?

`git` tracks changes on a line-by-line basis, so if you change a single character in a 5000 character-long paragraph written as a single line, it will highlight the entire line. Moreover, it will show it twice (original and changed version). And single line breaks are not a problem for Markdown... so,  there is really no benefit in using long lines as shorter ones make life much easier. 

Why exactly 100? This number is more-or-less arbitrary. People usually choose the number of characters that fit in their text editor without wrapping to the next (virtual) line. "100" is hard-coded into lesson-check and, honestly, I don't care about its exact value as long as it is reasonable: not too small, not too large... it could be 78, 83, 94, etc.

> Also, please ignore #519. I forgot to delete my old fork before I started work on the new file.

Um... ok, but we need to talk :) You should use branches for pull requests:

```
git checkout -f gh-pages
git pull origin gh-pages

git branch mybranch
git checkout mybranch
# make changes
git add -u
git commit
git push <your-fork>

git checkout gh-pages
# submit pull-request on GitHub
```

**Comment by doctornerdis** [comment]
Thanks for the explanation @maxim-belkin!

Don't I need permission to access this repository in order to make branches? I'm working from the GitHub desktop client, so do I need to work in `git` directly?

**Comment by maxim-belkin** [comment]
Um, to create branches in this repository one has to have proper permissions, but you don't need these.
The way it works is as follows:
1. you fork this repository. Your fork will be here https://github.com/doctornerdis/python-novice-inflammation
2. You clone your fork to your computer `git clone https://github.com/doctornerdis/python-novice-inflammation`
3. You create a new branch for your changes: `git checkout -b my-changes`
4. You make changes to files, then `git add -u`, `git commit`
5. You push your branch to your fork `git push`
6. You submit a PR on GitHub using this branch

To keep your fork up-to-date, you do:
```
git checkout gh-pages
git add swcarpentry https://github.com/swcarpentry/python-novice-inflammation  ## only once per repository
git fetch swcarpentry
git merge --ff-only
git push
```

Oh, and I now see that you've already deleted your fork that you used to submit #520. There is a better way! :)

**Comment by doctornerdis** [comment]
OK, I think I'm getting the hang of it now. I just did another PR (#521). Is that the right workflow?

**Comment by maxim-belkin** [comment]
Yep! Just don't forget to switch back to `gh-pages` before creating a branch for your next PR.

**Comment by MikeAllawayBham** [comment]
Hi @maxim-belkin , may be worth ticking off the lessons that have been done from the issue description at the top of this page, to make it clearer to newcomers which are left to do

**Comment by emdupre** [comment]
I just opened #565 to help address this ! I did have one question, though -- is the Code of Conduct still in need of definitions ? It seems to be rendering correctly, but I might have misunderstood the concern !

**Comment by maxim-belkin** [comment]
Thank you for the PR, @emdupre! I've updated the original post with the current list of issues. Regarding `Code of Conduct` - I think the issue was solved in `51293be`...

**Comment by katkoler** [comment]
Hi, I'd like to help with this. 
Is there a good way of testing it renders correctly?

**Comment by maxim-belkin** [comment]
Hi, @katkoler. If you've got macOS or Linux, `make serve` (or `jekyll serve` or `bundle exec jekyll serve`) with render the site for you and you will be able to check it out at `http://127.0.0.1:4000`.
If you're on a Windows `make docker-serve`.

**Comment by katkoler** [comment]
@maxim-belkin thanks! I'll just check that everything looks okay and open a pull request.

**Comment by SRSteinkamp** [comment]
Is there still any help needed?

**Comment by ldko** [comment]
Hi Simon,
Thank you for inquiring. I have updated the description with the output of running the command `$ make lesson-check-all` which shows that we have lines that needed shortening. If you would like to work on this issue, please edit one file per PR. Thanks!

**Comment by SRSteinkamp** [comment]
Thank you Idko for putting me into the loop :) 

I will work on the other files in the coming days, that's it for today :)  ... also if you have any comments on how to improve my PRs, I am happy to adjust them (especially in the future!)

**Comment by vinisalazar** [comment]
Hi!

I just ran the `make lesson-check-all` command and got the following output:

```
./CONTRIBUTING.md: Line(s) too long: 3, 114, 126
./README.md: Line(s) too long: 2, 4, 5, 7, 8, 9, 10, 13
./_episodes/03-matplotlib.md: Line(s) too long: 133
```

I will submit PRs to try to fix the remaining warnings.

Best,

V

**Comment by ldko** [comment]
Hi @vinisalazar , thanks for the offer! I believe the changes in CONTRIBUTING.md would need to be done outside of this lesson, specifically in the carpentries/style repo. The README.md lines that are too long are due to the markdown formatting for the table on the page, and I don't know of a way to get them under the 100 character mark and retain the formatting (if you do, please submit a fix :) ). Getting a PR from you for the 03-matlotlib.md file would be great!

**Comment by vinisalazar** [comment]
Hi @ldko ! Thank you for the fast reply.

Good observation. I'll have a look at the template repos to see if I can contribute something there.

Best,

V

**Comment by vinisalazar** [comment]
carpentries/lesson-example#330 addresses the `CONTRIBUTING.md` and `README.md` files.
